standalone/pruntime/enclave/src/contracts/btc_lottery.rs

```diff
@@ -180,12 +188,34 @@ impl BtcLottery {
                         return;
                     }
                 };
+                let secp = Secp256k1::new();
+                let public_key = PublicKey::from_private_key(&secp, &sk);
+                let prize_addr = Address::p2pkh(&public_key, Network::Bitcoin);
+                address_set.push(prize_addr.to_string().as_bytes().to_vec());
                 lottery_token.insert(String::from(winner_id), sk);
                 salt += 1;
             }
             self.lottery_set.insert(round_id, lottery_token);
             self.token_set.insert(round_id, round_token);
             self.round_id = round_id;
+
+            let payload = LotteryPayload::BtcAddresses { address_set };
+            let data = SendLottery {
+                round_id,
+                chain_id: 1,
+                token_id: Vec::new(),
```

Comment:
Token-id is only associated with the lottery opening event

Reply:
I moved the round_id and token_id into the `LotteryPayload` struct